xpublish_wms/wms/__init__.py

```diff
@@ -43,9 +44,9 @@ def wms_handler(
 
     match query:
         case WMSGetCapabilitiesQuery():
-            return get_capabilities(dataset, request, query)
+            return await asyncio.to_thread(get_capabilities, dataset, request, query)
```

Comment:
I don't really understand the context in which these other functions get called - is it important to make these properly async too?

Reply:
My guess is that it's only functions which actually trigger loading of additional chunk data which need to be fully async - if all they do is process metadata that should already be in memory then they don't need to be async.

Reply:
Once getmap is good, we need to make all the other methods also load data async instead of using to thread